### PR TITLE
Add agentic LoRA SFT smoke training

### LINKED ISSUE #692: OpenSearch-VL alignment: support agentic LoRA SFT - Milestone 3: verify base-vs-adapter behavior on trace fixtures - Unit 1: Add a tiny agentic fixture and smoke training path that can run locally.
## Executable Unit

Add a tiny agentic fixture and smoke training path that can run locally.

## Parent Context

- Direction: `OpenSearch-VL alignment: support agentic LoRA SFT`
- Plan/Milestone: `Milestone 3: verify base-vs-adapter behavior on trace fixtures`

## Definition Of Done

- The change is scoped to the smallest useful implementation slice.
- The governing spec or plan is created or updated before broad code changes.
- Tests cover the new behavior or contract edge case.
- Metrics and verification output are recorded in the PR evidence.

## Suggested Verification

- `git diff --check`
- Focused pytest or Swift test for the touched code path
- Changed-scope coverage for modified Python or Swift files when measurable
- Any relevant benchmark/evaluation artifact if this changes measured behavior

### PR DESCRIPTION
**Pull request by Zigfreidish**
## Summary
- Add a clean agentic LoRA SFT smoke fixture with one train and one validation trace.
- Add `scripts/agentic_lora_sft_smoke.py` to run deterministic `LoRATrainingPipeline` evidence without real MLX training.
- Document the governing #692 plan and update the trajectory contract anchors.

## Plan or Spec
- `docs/plans/2026-05-20-agentic-lora-sft-smoke-training.md`
- `docs/agentic-trajectory-dataset-contract.md`
- `docs/plans/2026-05-20-agentic-lora-sft-formatting.md`
- Closes #692

## Commands Run
```text
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_agentic_lora_sft_smoke.py services/mlx-worker-python/tests/test_agentic_sft_training_contract.py
Result: 6 passed.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_training_dataset_builder.py -k 'agentic or trajectory' services/mlx-worker-python/tests/test_agentic_lora_sft_smoke.py services/mlx-worker-python/tests/test_agentic_sft_training_contract.py
Result: 28 passed, 53 deselected.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage run --data-file=/tmp/agentic_lora_sft_smoke.coverage -m pytest -q services/mlx-worker-python/tests/test_agentic_lora_sft_smoke.py services/mlx-worker-python/tests/test_agentic_sft_training_contract.py
Result: 6 passed.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage json --data-file=/tmp/agentic_lora_sft_smoke.coverage -o /tmp/agentic_lora_sft_smoke_coverage.json
Result: wrote /tmp/agentic_lora_sft_smoke_coverage.json.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/python_changed_line_coverage.py --coverage-json /tmp/agentic_lora_sft_smoke_coverage.json scripts/agentic_lora_sft_smoke.py services/mlx-worker-python/tests/test_agentic_lora_sft_smoke.py
Result: TOTAL 99.33% (149/150).

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/agentic_lora_sft_smoke.py --json --output-dir .runtime/agentic-lora-sft-smoke-final
Result: passed; trainer_row_count=2, trainer_validation_row_count=2, response_only_boundary_count=4.

git diff --check
Result: passed.

make py-test
Result: 2868 passed, 14 skipped, 2 warnings.

git commit -m "Add agentic LoRA SFT smoke training"
Result: initial pre-commit attempt failed in make swift-test because the local disk was full during Swift text worker index writes; after clearing git-ignored build caches, retry passed make swift-test, make py-test, make integration-test, and the PR scoped performance report.

pre-commit make py-test
Result: 2868 passed, 14 skipped, 2 warnings.

pre-commit make integration-test
Result: 114 passed, 1 skipped.

pre-commit PR scoped performance report
Result: Status: ok; Changed files: 7; Selected probes: 0; Direct/gated probes: 0; Regressions: 0; Context regressions: 0; Verification failures: 0.
```

## Coverage and Metrics
- Changed-line coverage: 99.33% (149/150) across `scripts/agentic_lora_sft_smoke.py` and `services/mlx-worker-python/tests/test_agentic_lora_sft_smoke.py`.
- Smoke metrics: `trainer_row_count=2`, `trainer_validation_row_count=2`, `tool_call_count=2`, `observation_count=2`, `response_only_boundary_count=4`, `trace_tokens=83`, `tool_call_tokens=6`, `observation_tokens=23`, `final_answer_tokens=7`.
- Observability mode: evidence/offline smoke. Serving-path overhead: N/A, no request-serving code changes.
- Local pre-commit performance report: `Status: ok`, no direct/gated probes selected and no regressions.

## Known Gaps
- Real MLX-LM adapter training and base-vs-adapter quality comparison remain out of scope for #692 and are covered by follow-up #693.

### CODE REVIEW

File: scripts/agentic_lora_sft_smoke.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+
+def _read_jsonl(path: Path) -> list[dict[str, Any]]:
+    rows: list[dict[str, Any]] = []
+    if not path.is_file():
+        return rows
+    for raw_line in path.read_text(encoding="utf-8").splitlines():
+        line = raw_line.strip()
+        if line:
+            rows.append(json.loads(line))
```
[the remark this answers is not in the document]
Comment (Zigfreidish) [reply scripts/agentic_lora_sft_smoke.py]: Addressed in 774359fc: `_read_jsonl` now opens the JSONL file and iterates over the handle directly, so the smoke script no longer materializes the whole file before parsing rows.

File: services/mlx-worker-python/tests/test_agentic_lora_sft_smoke.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    with pytest.raises(FileNotFoundError):
+        agentic_lora_sft_smoke.run_smoke(REPO_ROOT, fixture_id="missing-fixture")
+
+
+def _jsonl_rows(path: Path) -> list[dict[str, object]]:
+    return [
+        json.loads(line)
+        for line in path.read_text(encoding="utf-8").splitlines()
+        if line.strip()
+    ]
```
[the remark this answers is not in the document]
Comment (Zigfreidish) [reply services/mlx-worker-python/tests/test_agentic_lora_sft_smoke.py]: Addressed in 774359fc: the test helper now iterates over the opened JSONL file handle directly before parsing non-empty rows.
